app/graphql/types/concerns/profile_scoring.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module Types
+  # Methods related to profile scoring
+  module ProfileScoring
+    extend ActiveSupport::Concern
+
+    def compliant_host_count
+      ::CollectionLoader.for(
+        object.class,
+        object.policy_id ? :policy_test_result_hosts : :test_result_hosts
+      ).load(object).then do |hosts|
+        hosts.count { |host| object.compliant?(host) }
+      end
+    end
+
+    def total_host_count
+      ::CollectionLoader.for(policy_or_report.class, :hosts)
+                        .load(policy_or_report).then(&:count)
+    end
+
+    def test_result_host_count
+      ::CollectionLoader.for(
+        object.class,
+        object.policy_id ? :policy_test_result_hosts : :test_result_hosts
+      ).load(object).then(&:count)
```

Comment:
This does not count hosts distinctively. I had a host reporting under with SSGs (before there was a test result auto-remove) and now it reports as two hosts. I'll never get 100%, unless I clean it manually. 

I guess we should not depend on the test-result auto-remove, and always count it distinctively. There might be the same problem with the compliant host count (if I did have the system compliant for both reporting SSGs).

Reply:
TIL we can add `-> { distinct }` to our relationships. I'll be adding it to `test_result_hosts` and `policy_test_result_hosts`.

Reply:
Ok, so adding `distinct` here fixes the host total counts, but I'm not sure how we should handle the score in the UI. It looks like we're computing an average of all SSG versions, which is definitely _not_ what we want to do. The following host has two reports from different SSG versions, with scores of 95 and 45:

![image](https://user-images.githubusercontent.com/761923/99681987-c0797b00-2a4c-11eb-8b79-b906b9fe5e13.png)

Also, the number of compliant systems needs to be fixed for this scenario still.

Reply:
Looking at the screenshot above, I realized it's not listing my other system! That system has a report for this policy but was unassigned from it. I'll need to fix it. Here I've re-assigned the second host to the policy:

![image](https://user-images.githubusercontent.com/761923/99682485-41d10d80-2a4d-11eb-9230-18a784aa2eb5.png)